alacritty_terminal/src/term/mod.rs

```diff
@@ -2301,6 +2326,52 @@ mod tests {
         scrolled_grid.scroll_display(Scroll::Top);
         assert_eq!(term.grid, scrolled_grid);
     }
+
+    #[test]
+    fn should_not_exit_if_hold() {
+        let size = SizeInfo {
+            width: 21.0,
+            height: 51.0,
+            cell_width: 3.0,
+            cell_height: 3.0,
+            padding_x: 0.0,
+            padding_y: 0.0,
+            dpr: 1.0,
+        };
+        let mut config = MockConfig::default();
+        config.hold = true;
+        let mut term = Term::new(
+            &config,
+            &size,
+            Clipboard::new_nop(),
+            EventSpy { latest_event: RefCell::new(None) },
+        );
+        term.handle_child_process_exit();
+        assert_eq!(term.event_proxy.latest_event.into_inner(), None);
+    }
+
+    #[test]
+    fn should_exit_if_not_hold() {
+        let size = SizeInfo {
+            width: 21.0,
+            height: 51.0,
+            cell_width: 3.0,
+            cell_height: 3.0,
+            padding_x: 0.0,
+            padding_y: 0.0,
+            dpr: 1.0,
+        };
+        let mut config = MockConfig::default();
+        config.hold = false;
+        let mut term = Term::new(
+            &config,
+            &size,
+            Clipboard::new_nop(),
+            EventSpy { latest_event: RefCell::new(None) },
+        );
+        term.handle_child_process_exit();
+        assert_eq!(term.event_proxy.latest_event.into_inner(), Some(Event::Exit));
+    }
```

Comment:
Nice tests :tada:

We should really have a faster way to build a test `Terminal`...